### PR TITLE
:seedling: Add test for scale testing machinery

### PR DESCRIPTION
**Pull request by killianmuldoon**
A test to ensure the scale test is running before adding to periodic.

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
/test

**Review by killianmuldoon — COMMENTED** [review]
/area e2e-testing

File: test/e2e/scale_test.go [hunk 1]
```diff
@@ -24,7 +24,26 @@ import (
 	"k8s.io/utils/pointer"
 )
 
-var _ = Describe("When scale testing using in-memory provider [Scale]", func() {
+var _ = Describe("When testing the machinery for scale testing using in-memory provider", func() {
```
Comment (killianmuldoon) [inline test/e2e/scale_test.go:27]: This is a duplicate in terms of spec, but the point is to have a test with no-test so we're testing e2e for the in-memory provider to ensure it's up and running.


This test should run as part of e2e-full while leaving the [Scale] tag for future work around scale testing.

**Comment by killianmuldoon** [comment]
@fabriziopandini - seeing as you had the idea originally @https://vmware.slack.com/archives/C0523T5H7MF/p1694765812754249

**Review by chrischdi — COMMENTED** [review]
/lgtm

**Pull request merged**